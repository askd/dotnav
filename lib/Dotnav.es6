import React from 'react';
import PropTypes from 'prop-types';

export default class Dotnav extends React.Component {
  render() {
    const { withArrows, count, index, classes, handleChange } = this.props;

    if (count < 2) return null;

    return (
      <nav className = { classes.root }>
        { withArrows && <button
          type = 'button'
          className = { classes.arrowPrev }
          onClick = { () => handleChange((index <= 0) ? count - 1 : index - 1) }
        /> }
        <ul className = { classes.dots }>
          { Array.from({ length: count }, (_, i) =>
            <li
              key = { i }
              className = { index === i ? classes.dotActive : classes.dot }
              onClick = { () => handleChange(i) }
            />
          )}
        </ul>
        { withArrows && <button
          type = 'button'
          className = { classes.arrowNext }
          onClick = { () => handleChange((index >= count - 1) ? 0 : index + 1) }
        /> }
      </nav>
    );
  }
}

Dotnav.propTypes = {
  withArrows: PropTypes.bool,
  count: PropTypes.number,
  index: PropTypes.number,
  handleChange: PropTypes.func,
  classes: PropTypes.shape({
    root: PropTypes.string,
    dots: PropTypes.string,
    dot: PropTypes.string,
    dotActive: PropTypes.string,
    arrowPrev: PropTypes.string,
    arrowNext: PropTypes.string,
  }),
};

Dotnav.defaultProps = {
  withArrows: false,
  count: 0,
  index: 0,
  handleChange: () => {},
  classes: {
    root: 'dotnav',
    dots: 'dotnav-dots',
    dot: 'dotnav-dot',
    dotActive: 'dotnav-dot dotnav-dot_active',
    arrowPrev: 'dotnav-arrow dotnav-arrow_prev',
    arrowNext: 'dotnav-arrow dotnav-arrow_next',
  },
};
