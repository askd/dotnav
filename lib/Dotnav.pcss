.Dotnav {
  position: relative;
  padding: 10px 40px;
  border-radius: 20px;
  background-color: #000;
}

.Dotnav-dots {
  display: flex;
  justify-content: center;
}

.Dotnav-dot {
  width: 10px;
  height: 10px;
  margin: 0 5px;
  padding: 5px;
  cursor: pointer;
}

.Dotnav-dot::after {
  content: '';
  display: block;
  width: 10px;
  height: 10px;
  border-radius: 50%;
  background-color: #fff;
  transition: box-shadow 0.5s;
}

.Dotnav-dot:hover::after,
.Dotnav-dot:focus::after {
  box-shadow: 0 0 0 3px color(#fff a(50%));
}

.Dotnav-dot_active {
  cursor: default;
}

.Dotnav-dot_active::after,
.Dotnav-dot_active:hover::after,
.Dotnav-dot_active:focus::after {
  box-shadow: 0 0 0 3px #fff;
}

.Dotnav-arrow {
  position: absolute;
  top: 50%;
  width: 40px;
  height: 40px;
  margin: -20px 0 0 0;
  padding: 0;
  border: none;
  border-radius: 20px;
  box-shadow: none;
  outline: none;
  background-color: transparent;
  color: #fff;
  cursor: pointer;
  transition: background-color 0.5s;
}

.Dotnav-arrow:focus,
.Dotnav-arrow:hover {
  background-color: rgba(255, 255, 255, 0.25);
}

.Dotnav-arrow::before,
.Dotnav-arrow::after {
  content: '';
  position: absolute;
  top: 19px;
  left: 10px;
  width: 20px;
  height: 5px;
  border-radius: 3px;
  background-color: currentColor;
}

.Dotnav-arrow::before {
  transform: rotate(-45deg);
}

.Dotnav-arrow::after {
  transform: rotate(45deg);
}

.Dotnav-arrow_prev {
  left: 0;
}

.Dotnav-arrow_prev::after {
  margin-top: -3px;
}

.Dotnav-arrow_prev::before,
.Dotnav-arrow_prev::after {
  transform-origin: 0% 50%;
}

.Dotnav-arrow_next {
  right: 0;
}

.Dotnav-arrow_next::before {
  margin-top: -3px;
}

.Dotnav-arrow_next::before,
.Dotnav-arrow_next::after {
  transform-origin: 100% 50%;
}
