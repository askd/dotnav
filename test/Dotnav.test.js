/* eslint-env jest */

import { shallow } from 'enzyme';

import React from 'react';
import Dotnav from '../lib/Dotnav.js';

describe('<Dotnav />', () => {
  it('should render', () => {
    const component = shallow(<Dotnav withArrows count={3} index={0} />);

    expect(component).toMatchSnapshot();
  });
});
