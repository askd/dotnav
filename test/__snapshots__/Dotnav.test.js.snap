// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dotnav /> should render 1`] = `
<nav
  className="Dotnav"
>
  <button
    className="Dotnav-arrow Dotnav-arrow_prev"
    onClick={[Function]}
    type="button"
  />
  <ul
    className="Dotnav-dots"
  >
    <li
      className="Dotnav-dot Dotnav-dot_active"
      onClick={[Function]}
    />
    <li
      className="Dotnav-dot"
      onClick={[Function]}
    />
    <li
      className="Dotnav-dot"
      onClick={[Function]}
    />
  </ul>
  <button
    className="Dotnav-arrow Dotnav-arrow_next"
    onClick={[Function]}
    type="button"
  />
</nav>
`;
